plugins/org.eclipse.reddeer.eclipse/src/org/eclipse/reddeer/eclipse/wst/server/ui/cnf/ServerModule.java

```diff
@@ -185,5 +191,46 @@ public void restart(){
 		Class<?>[] classes = {TreeItem.class, ServersView2.class};
 		return classes;
 	}
+	
+	/**
+	 * Get direct child modules
+	 * 
+	 * @return List of modules, which are direct module children
+	 */
+	public List<ServerModule> getModules() {
+		List<ServerModule> childModules = new ArrayList<ServerModule>();
+		List<TreeItem> childModuleTreeItems = this.treeItem.getItems();
+		for (TreeItem item : childModuleTreeItems) {
+			childModules.add(new ServerModule(item, view));
+		}
+		return childModules;
+	}
+	
+	/**
+	 * Returns submodule's name
+	 * 
+	 * @return the submodule's name
+	 */
+	public String getName() {
+		return this.treeItem.getText().split(" ")[0];
```

Comment:
Why you dont use your newly implemented getTreeItem()?

Reply:
Fixed